### PR TITLE
Start tidying kubeadm up

### LINKED ISSUE #33916: kubeadm deploys kube-proxy daemonsets for different architectures, but it doesn't work
Right now we create 3 daemonsets for `kube-proxy` in `app/master/addons.go`, but `images.GetCoreImage()` will only give use image names for `runtime.GOARCH`, so this feature is broken. We should only deploy one daemon set for the native architecture, but ensure it has the right label selectors, so we don't try to run `amd64` pods on nodes of different architecture. We should also make sure all addons have architecture labels, and not just `kube-proxy`. If user is building heterogeneous clusters, the will need to take about deploying addons for additional architectures, we shouldn't pre-deploy multiple things knowing that they will most likely remain unschedulable.

### PR DESCRIPTION
**Pull request by errordeveloper**
**What this PR does / why we need it**:

This PR addresses https://github.com/kubernetes/kubernetes/pull/33262#discussion_r80337511.

**Release note**:

``` release-note
NONE
```

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35119)

### CODE REVIEW

**Comment by pires** [comment]
As said on Slack, this PR should be split into 1) multi-arch changes 2) tidying up. However, tidying up/refactoring stuff is something I look at very cautiously because we have 0 tests.

**Comment by errordeveloper** [comment]
@pires ok, I broke this up.

**Comment by errordeveloper** [comment]
The multi-arch fix is in https://github.com/kubernetes/kubernetes/pull/35124 now.

**Comment by errordeveloper** [comment]
@k8s-bot cvm gke e2e test this

**Comment by errordeveloper** [comment]
@k8s-bot cvm gke e2e test this

**Comment by mikedanese** [comment]
Since it addresses @krousey 's review comment :)

File: cmd/kubeadm/app/master/manifests.go [hunk 1]
```diff
@@ -221,88 +221,100 @@ func componentPod(container api.Container, volumes ...api.Volume) api.Pod {
...
-		proxy: {},
-	}
-
+func getComponentBaseCommand(component string) (command []string) {
 	envParams := kubeadmapi.GetEnvParams()
 	if envParams["hyperkube_image"] != "" {
 		command = []string{"/hyperkube", component}
 	} else {
-		command = []string{"/usr/local/bin/kube-" + component}
+		command = []string{"kube-" + component}
```
Comment (krousey) [inline cmd/kubeadm/app/master/manifests.go]: Why was this leading path removed? Should the path to hyperkube above still be absolute?

Comment (errordeveloper) [inline cmd/kubeadm/app/master/manifests.go]: No, it's not in the path...

**Comment by errordeveloper** [comment]
@k8s-bot gci gke e2e test this

**Comment by errordeveloper** [comment]
@k8s-bot unit test this

**Comment by errordeveloper** [comment]
@k8s-bot unit test this

**Comment by errordeveloper** [comment]
@k8s-bot gci gke e2e test this

**Comment by pires** [comment]
@errordeveloper this has been merged already.

**Pull request merged**